File: gitlfs-common/src/main/java/ru/bozaro/gitlfs/common/data/BatchRes.java
package ru.bozaro.gitlfs.common.data;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import javax.annotation.Nonnull;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * Batch request.
 *
 * @author Artem V. Navrotskiy
 */
public final class BatchRes {
  @JsonProperty(value = "objects", required = true)
  @Nonnull
  private final List<BatchItem> objects;

  @JsonCreator
  public BatchRes(
      @JsonProperty(value = "objects", required = true) @Nonnull List<BatchItem> objects
  ) {
    this.objects = Collections.unmodifiableList(new ArrayList<>(objects));
  }

  @Nonnull
  public List<BatchItem> getObjects() {
    return objects;
  }
}

File: gitlfs-common/src/main/kotlin/ru/bozaro/gitlfs/common/data/BatchRes.kt
package ru.bozaro.gitlfs.common.data

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*

/**
 * Batch request.
 *
 * @author Artem V. Navrotskiy
 */
class BatchRes @JsonCreator constructor(
        @JsonProperty(value = "objects", required = true) objects: List<BatchItem>
) {
    @JsonProperty(value = "objects", required = true)
    val objects: List<BatchItem> = Collections.unmodifiableList(ArrayList(objects))
}
